keps/sig-apps/0753-sidecarcontainers.md

```diff
@@ -235,23 +234,25 @@ will log and _logging container_ the sidecar that will facilitate it.
 
 If the logging container starts after the main app, some logs can be lost.
 Furthermore, if the logging container is not yet started and the main app
-crashes on startup, those logs can be lost (depends if logs to a shared volume
-or over the network on localhost, etc.). Startup is usually not that critical,
-because if the case where the other container is not started (like a restart) is
-important, is usually handled correctly.
-
-On shutdown the ordering behavior is arguably more important: if the logging
+crashes on startup, those logs can be lost (depends if logs go to a shared volume
+or over the network on localhost, etc.).
+// I don't really understand this sentence
+Startup is usually not that critical,
+because in the case where the other container is not started (like a restart) is
+important, it is usually handled correctly.
```

Comment:
I wanted to call this out as I read this sentence a few times and I wasn't really sure what it meant or what you were trying to convey.

Reply:
Oh, what I was trying to say is that the startup is probably less problematic for a logs or metrics sidecar.

If the app cares about not losing logs, for example, it will handle the case when the sidecar crashes to not lose anything. That same retry will work in case the app is started before the sidecar.

That is what I meant. Now that I read it with fresh eyes, doesn't seem very clear. Do you have any suggestion on how to say that in a clear way? :)

Reply:
Right, so you're saying that some people will have designed their apps to deal with this scenario? While that may be true for some applications for a generic application it might not be. I'd maybe call out that some applications will be built to handle these scenarios but this may require code changes which are not always possible.

Reply:
No, not trying to say that, but the opposite.

I was trying to say: well, I know people may say this is not so important, because if it is important it should already be handled for the restart case. However, although that may make some people think this is not so important, the shutdown problem is something even harder to argue against.

I was trying to say "ok, you may think apps needs to change for that, but even if they do, the shutdown is not possible". And some other examples (like service mesh) show why startup order is important too.

Does it makes sense? How would you phrase it?

Reply:
Maybe something like
"While you can modify your application to handle this scenario during startup, for shutdown this approach won't work"

Reply:
heh, I really like the way you can make very short what takes me several sentences :-D

Will use that, thanks! :)

Reply:
Fixed it here: https://github.com/kinvolk/kubernetes-enhancements/commit/a7465c9cb912015bfd72eb636bd058278abff110#diff-a08bdac8124fe291c1085aeb90981019

Will amend if it looks fine to you!